sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -159,10 +158,7 @@ case class UnresolvedAttribute(nameParts: Seq[String]) extends Attribute with Un
 
   override def toString: String = s"'$name"
 
-  override def sql: String = name match {
-    case ParserUtils.escapedIdentifier(_) | ParserUtils.qualifiedEscapedIdentifier(_, _) => name
-    case _ => quoteIdentifier(name)
-  }
+  override def sql: String = nameParts.map(quoteIdentifier(_)).mkString(".")
```

Comment:
since we are touching it, how about we update `quoteIdentifier` to make it match `CatalogV2Implicits.quoteIfNeeded`?

Reply:
Thanks. I tried to replace `quoteIdentifier` with `quoteIfNeeded`.
Seems to work fine.